test/integration/specs/task.js

```diff
@@ -387,6 +387,28 @@ describe('/:task', () => {
         });
     });
 
+    it('adds a request id to the transaction and persists through all steps in the request', () => {
+      const payload = { status: 'first', meta: { comment: 'some reason' } };
+      this.flow.hook('status:*:first', m => m.setStatus('second'));
+      this.flow.hook('status:*:second', m => m.setStatus('third'));
+      return request(this.app)
+        .put(`/${id}/status`)
+        .set('Content-type', 'application/json')
+        .send(payload)
+        .expect(200)
+        .then(() => {
+          return request(this.app)
+            .get(`/${id}`)
+            .expect(200)
+            .expect(response => {
+              assert.equal(response.body.data.status, 'third', 'The status should have been updated to third');
+              const activityLog = response.body.data.activityLog;
+              const ids = activityLog.map(log => log.event.req).filter(Boolean);
```

Comment:
This `filter` smells bad to me. If there's activity that doesn't get a request id then that should fail the test.

Reply:
yeah it's because we add a comment before this test